compiler/rustc_span/src/hygiene.rs

```diff
@@ -572,35 +578,32 @@ impl HygieneData {
         }
 
         // Reserve a new syntax context.
-        // The inserted dummy data can only be potentially accessed by nested `alloc_ctxt` calls,
-        // the assert below ensures that it doesn't happen.
         let ctxt = SyntaxContext::from_usize(self.syntax_context_data.len());
         self.syntax_context_data
             .push(SyntaxContextData { dollar_crate_name: sym::dummy, ..SyntaxContextData::root() });
         self.syntax_context_map.insert(key, ctxt);
 
-        // Opaque and semi-opaque versions of the parent. Note that they may be equal to the
-        // parent itself. E.g. `parent_opaque` == `parent` if the expn chain contains only opaques,
-        // and `parent_opaque_and_semiopaque` == `parent` if the expn contains only (semi-)opaques.
```

Comment:
Could you please keep all the comments here and below.
I https://github.com/rust-lang/rust/pull/139083 I really tried to make the logic here understandable, but this PR not only makes it more complex but also removes comments.